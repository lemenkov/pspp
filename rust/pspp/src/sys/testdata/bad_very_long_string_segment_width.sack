# File header.
"$FL2"; s60 "$(#) SPSS DATA FILE PSPP synthetic test file";
2; 1; 1; 0; -1; 100.0; "01 Jan 11"; "20:53:52"; s64 ""; i8 0 *3;

# Variables.
2; 255; 0; 0; 0x01ff00 *2; s8 "STR1";
(2; -1; 0; 0; 0; 0; s8 "") * 31;
2; >>9<<; 0; 0; 0x010900 *2; s8 "STR1_A";
>>2; -1; 0; 0; 0; 0; s8 "";<<

# Very long string map.
7; 14; 1; COUNT (
"STR1=00256"; i8 0; i8 9;
);

# Character encoding record.
7; 20; 1; 12; "windows-1252";

# Dictionary termination record.
999; 0;
