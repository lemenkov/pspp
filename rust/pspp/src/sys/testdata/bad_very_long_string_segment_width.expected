File header claims 1 variable positions but 34 were read from file.

Variable with short name STR1 listed in very long string record with width 256 has segment 1 of width 9 (expected 4).

╭──────────────────────┬────────────────────────╮
│       Created        │    01-JAN-2011 20:53:52│
├──────────────────────┼────────────────────────┤
│Writer Product        │PSPP synthetic test file│
├──────────────────────┼────────────────────────┤
│       Compression    │SAV                     │
│       Number of Cases│Unknown                 │
╰──────────────────────┴────────────────────────╯

╭─────────┬─╮
│Variables│2│
╰─────────┴─╯

╭──────┬────────┬─────┬─────────────────┬─────┬─────┬─────────┬────────────┬────────────┬──────────────╮
│      │Position│Label│Measurement Level│ Role│Width│Alignment│Print Format│Write Format│Missing Values│
├──────┼────────┼─────┼─────────────────┼─────┼─────┼─────────┼────────────┼────────────┼──────────────┤
│str1  │       1│     │Nominal          │Input│   32│Left     │A255        │A255        │              │
│str1_a│       2│     │Nominal          │Input│    9│Left     │A9          │A9          │              │
╰──────┴────────┴─────┴─────────────────┴─────┴─────┴─────────┴────────────┴────────────┴──────────────╯
